api/sessionrecording/session_recording.go

```diff
@@ -99,13 +120,14 @@ type ReaderStats struct {
 	TotalEvents int64
 }
 
-// ToFields returns a copy of the stats to be used as log fields
-func (p ReaderStats) ToFields() map[string]any {
-	return map[string]any{
-		"skipped-events":      p.SkippedEvents,
-		"out-of-order-events": p.OutOfOrderEvents,
-		"total-events":        p.TotalEvents,
-	}
+// LogValue returns a copy of the stats to be used as log fields
+func (p ReaderStats) LogValue() slog.Value {
+	return slog.GroupValue(
+		slog.Int64("skipped_bytes", p.SkippedBytes),
+		slog.Int64("skipped_events", p.SkippedEvents),
+		slog.Int64("out_of_order_events", p.OutOfOrderEvents),
+		slog.Int64("total_events", p.TotalEvents),
+	)
 }
 
 // Close releases reader resources
```

Comment:
rosstimothy brought up the following:

"Is is possible that gzipReader is assigned after the check on line 129? For instance what if Read is being executed in another goroutine?"